omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -0,0 +1,245 @@
+					stopModules(modules, servletContext);
+					break;
+				case RESTART:
+					restartModules(modules, servletContext);
+					break;
+				case UNLOAD:
+					unloadModules(modules, servletContext);
+					break;
+			}
+		}
+		
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private ModuleAction parseAction(SimpleObject post, RequestContext context) {
+		
+		//parse action
+		String actionTypeName = post.get("action");
+		ModuleAction.Type actionType;
+		try {
+			if (actionTypeName == null) {
+				throw new IllegalArgumentException("Please specify action type");
+			}
+			actionType = ModuleAction.Type.valueOf(actionTypeName.toUpperCase().trim());
+		}
+		catch (Exception e) {
+			throw new IllegalArgumentException("Invalid modules action", e);
+		}
+		
+		//parse modules
+		List<String> moduleUUIDs = post.get("modules");
+		Collection<Module> modules = new ArrayList<Module>();
+		if (moduleUUIDs != null) {
+			for (String bodyModule : moduleUUIDs) {
+				modules.add(findByUuid(bodyModule));
+			}
+		}
+		
+		boolean allModules = false;
+		Object allModulesValue = post.get("allModules");
+		
+		//instanceof returns false if null
+		if (allModulesValue instanceof Boolean) {
+			allModules = (Boolean) allModulesValue;
+		} else if (allModulesValue instanceof String) {
+			allModules = Boolean.valueOf((String) allModulesValue);
+		}
+		
+		if (allModules && !modules.isEmpty()) {
+			throw new IllegalArgumentException("Please choose either 'allModules' or modules");
+		}
+		return new ModuleAction(actionType, modules, allModules);
+	}
+	
+	private void restartModules(Collection<Module> modules, ServletContext servletContext) {
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+			}
+		}
+		
+		boolean needsRefresh = false;
+		for (Module module : moduleFactoryWrapper.getModulesInStartupOrder(modules)) {
+			if (moduleFactoryWrapper.isModuleStopped(module)) {
+				needsRefresh = moduleFactoryWrapper.startModuleSkipRefresh(module, servletContext) || needsRefresh;
+			}
+		}
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	private void unloadModules(Collection<Module> modules, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+				needsRefresh = true;
+			}
+			moduleFactoryWrapper.unloadModule(module);
+		}
+		
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	private void stopModules(Collection<Module> modules, ServletContext servletContext) {
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModule(module, servletContext);
```

Comment:
The above refreshes the spring application context for each module. Was this your intention?